### PR TITLE
Introduce property to provide options for IAsyncEnumerable conversion

### PR DESCRIPTION
**Pull request by ValerieFernandes**
Issue: #19

Added an optional property to the CreateSyncVersionAttribute to specify which of IList, ReadOnlySpan, Span along side or instead of IEnumerable an IAsyncEnumerable, allows for multiple conversions.

A few notes that could help with reviewing:
- I added the new conversion (e.g. "IAsyncEnumerable", "IList") to a dictionary that then overwrites the default replacement. Right now dictionary is unnecessary, but this solution will support any amount of conversion overrides if that is potentially useful.
- The tests returned methods as part of partial class when multiple overrides occurred I wasn't sure if that is correct behaviour or not so I kept it in the verified solution for now.

### CODE REVIEW

File: src/Zomp.SyncMethodGenerator/SyncMethodSourceGenerator.cs [hunk 1]
```diff
@@ -120,13 +121,19 @@ private static List<MethodToGenerate> GetTypesToGenerate(SourceProductionContext
 
             var methodName = methodSymbol.ToString();
 
+            var variations = 8;
```
Comment (virzak) [inline src/Zomp.SyncMethodGenerator/SyncMethodSourceGenerator.cs:124]: Set to CollectionTypes.IEnumrable

File: src/Zomp.SyncMethodGenerator/SourceGenerationHelper.cs [hunk 2]
```diff
@@ -7,14 +7,38 @@ public static class SourceGenerationHelper
 {
     internal const string CreateSyncVersionAttributeSource = """
 // <auto-generated/>
+using System;
```
[the remark this answers is not in the document]
Comment (ValerieFernandes) [reply src/Zomp.SyncMethodGenerator/SourceGenerationHelper.cs:10]: When I remove it it cant recognize the FlagsAttribute on the enum

File: src/Zomp.SyncMethodGenerator/SourceGenerationHelper.cs [hunk 3]
```diff
@@ -7,14 +7,38 @@ public static class SourceGenerationHelper
...
+
     /// <summary>
     /// An attribute that can be used to automatically generate a synchronous version of an async method. Must be used in a partial class.
     /// </summary>
     [System.AttributeUsage(System.AttributeTargets.Method)]
     internal class CreateSyncVersionAttribute : System.Attribute
     {
+        private CollectionTypes variations;
+
+        public virtual CollectionTypes Variations
```
[the remark this answers is not in the document]
Comment (ValerieFernandes) [reply src/Zomp.SyncMethodGenerator/SourceGenerationHelper.cs]: changed it as requested, original way of doing it was from here: https://learn.microsoft.com/en-us/dotnet/standard/attributes/writing-custom-attributes#declaring-properties

File: src/Zomp.SyncMethodGenerator/SourceGenerationHelper.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (virzak) [reply src/Zomp.SyncMethodGenerator/SourceGenerationHelper.cs:10]: See https://github.com/zompinc/sync-method-generator/issues/13#issuecomment-1809214590

So we should either use global::System.Flags or move it inside the namespace.

**Pull request closed without merging**